CraftTweaker2-MC1120-Mod-JEI/src/main/java/crafttweaker/mods/jei/actions/DescribeAction.java

```diff
@@ -31,7 +32,7 @@ public DescribeAction(IItemStack stack, String[] description) {
 	 * @param name Solely for the User output, should be a representation of stack 
 	 */
 	public DescribeAction(List<? extends IIngredient> stack, String[] description, String name) {
-		this.stack = stack.stream().map(IIngredient::getInternal).collect(Collectors.toList());
+		this.stack = stack.stream().map(IIngredient::getInternal).map(o -> (Ingredient) o).collect(Collectors.toList());

```

Comment:
Is this tested?
Looks like there'd be ClassCastException in the map function 🤔 

`IItemStack#getInternal` returns an ItemStack, and that is not an Ingredient iirc.
`ILiquidStack#getInternal` returns a FluidStack, and that is not an Ingredient either iirc.

You could use `CraftTweakerMC::getIngredient` or something.
However, that would not work for the fluids either, since for describing you want the fluidStack.
You could go for two classes then, one for fluids, one for items or something, up to you.

Reply:
:thinking: But the `stack` variable takes a List with Ingredient before this commit, so the method would fail anyway?

Reply:
No.
For example, say `stack` was `List<ILiquidStack>`
Then, `this.stack` would (effectively) be `List<FluidStack>`

but when `stack` was `List<IItemStack>`
Then, `this.stack` would (effectively) be `List<ItemStack>`

Reply:
The map() call is what would error

Reply:
but `DescribeAction#stack` is already defined as `List<? extends Ingredient>` - neither `FluidStack` nor `ItemStack` extend `Ingredient`

Reply:
That parameter that is provided is `IIngredient` (CrT) not `Ingredient` (MC).
If you check the calls of this constructor, you'll see that it is only ever called with either a list of IItemStack, or a SingletonList of IFluidStack.

The instance variable `stack` on the other hand orignally was a list of Object, which works, since with these two calls it would be either a list of ItemStack (MC), or FluidStack(MC)

Reply:
oh my bad - i have no idea why it changed to `List<? extends Ingredient>`

Reply:
Actually, it's probably because the `addIngredientInfo` now can't differentiate between the first parameter as `Object` or `List<Object>`, which I'm not sure if it'd cause any issues or if it'd select the `List` because it's less permissive.

Reply:
Linter still says `Object` will be used, but I guess it'd use `List<?>` at runtime.